### PR TITLE
BUG: SeriesGroupBy.value_counts() raising error on an empty Series

### LINKED ISSUE #39172: BUG: SeriesGroupBy.value_counts() raises on an empty Series
- [X] I have searched the [[pandas] tag](https://stackoverflow.com/questions/tagged/pandas) on StackOverflow for similar questions.

- [ ] I have asked my usage related question on [StackOverflow](https://stackoverflow.com).

---

#### Question about pandas

Using the `value_counts` method after grouping an empty DataFrame and selecting a column raises an error.

Example:

```python
>>> pd.DataFrame(columns=["A", "B"]).groupby("A")["B"].value_counts()
```

Error:

```python-traceback
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/username/.virtualenvs/my_project/lib/python3.8/site-packages/pandas/core/groupby/generic.py", line 736, in value_counts
    codes = [rep(level_codes) for level_codes in codes] + [llab(lab, inc)]
  File "/Users/username/.virtualenvs/my_project/lib/python3.8/site-packages/pandas/core/groupby/generic.py", line 705, in <lambda>
    llab = lambda lab, inc: lab[inc]
IndexError: boolean index did not match indexed array along dimension 0; dimension is 0 but corresponding boolean dimension is 1
```

Instead, I would expect that `value_counts` returns an empty series, like when using the `max` method, like the following,

```python
>>> pd.DataFrame(columns=["A", "B"]).groupby("A")["B"].max()
```

which gives this result:

```
Series([], Name: B, dtype: object)
```

If it is normal to have an error when using `value_counts` for an empty `SeriesGroupBy`, why is it so?

Thank you.

### PR DESCRIPTION
**Pull request by AnnaDaglis**
- [x] closes #39172
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -731,10 +731,14 @@ def apply_series_value_counts():
 
         # group boundaries are where group ids change
         idx = np.r_[0, 1 + np.nonzero(ids[1:] != ids[:-1])[0]]
+        if len(ids) == 0:
```
Comment (jreback) [inline pandas/core/groupby/generic.py:734]: can you do `not len(ids)` (and same below)

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm @AnnaDaglis minor comment, pls ping on green

**Comment by AnnaDaglis** [comment]
@jreback Green and ready to go!

**Comment by jreback** [comment]
looks good. can you merge master and ping on green (the reason is just merged a PR that changes the hashing of .value_counts()); it should be fine, just making sure.

**Comment by AnnaDaglis** [comment]
@jreback I've merged the master, and it's green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @AnnaDaglis keep em coming!

**Pull request merged**